### DISCUSSION TITLE
Questions about code quality, code expectations and code compromises.

### DISCUSSION THREAD

**Discussion by wreper12** [tteck/Proxmox#2753] [Q&A]
Hello, hopefully your day is going well. I have been working on a Proxmox VE script and I had some question about how the code should be formatted and setup.

1. Do you have any resources to show how you want the code to be formatted and how the structure of the code should be and information about good and bad practices which you want the BASH scripts to maintain?

2. Are there any specific things that should be done in the Proxmox script or anything that should be specifically avoided? 

3. If a program offers 2 ways to install where one option is more reliable but leads to a suboptimal situation (Running docker in a LXC for example), which option would be preferred? I have seen many programs where when installed locally they can have problems whereas the docker version works with very few problems.

**Comment by tteck** [comment]
1) Review the existing scripts in the repository to understand the preferred formatting and structural conventions, ensuring consistency and adherence to established practices.

2) Leverage resources like https://www.patorjk.com/software/taag/#p=display&f=Slant&t=Helper-Scripts for generating ASCII art, maintaining a standardized visual representation across scripts.

3) Prioritize native LXC installations, Docker application scripts will not be accepted.

**Comment by tteck** [comment]
> leads to a suboptimal situation (Running docker in a LXC for example),

I have never encountered any problems when using Docker within an LXC.

**Reply by wreper12** [reply]
Sorry I wrote that poorly, I meant that it is suboptimal to have multiple virtualization layers.

**Comment by DrEVILish** [comment]
You can read the Dockerfile to work out what it does to install the program.
This would be the best way to move from Docker to LXC, I've done this for a couple of LXC containers I'm running. 
Read the docker file and copied the install procedure.
